src/parser.rs

```diff
@@ -67,10 +70,27 @@ fn parse_stmt(ts: &[Token]) -> Option<(Stmt, &[Token])> {
 }
 
 fn parse_expr(ts: &[Token]) -> Option<(Expr, &[Token])> {
+    if let [Token::Path(s), ..] = ts {
+        let mut exprs: Vec<Expr> = Vec::new();
+        let mut ts = &ts[1..];
+        loop {
+            // Gather the following expressions as a vector for modifying the command.
+            if let Some((expr, ts0)) = parse_factor(ts) {
+                exprs.push(expr);
+                ts = ts0;
+                continue;
+            }
+            break;
+        }
+        let expr = Expr::Cmd(Box::new(Expr::Path(String::from(s))), exprs);
```

Comment:
```rust
Expr::Cmd(box Expr::Path(String::from(s)), exprs);
```

Reply:
Maybe you didn't do this because of more parens 🤷

Reply:
I think I was following the precedent that was set in lines 58-63 for parse_stmt. I assumed you did this for readability so i did the same.
`if let [Token::Var(name), Token::Eq, ..] = ts {
        if let Some((expr, ts)) = parse_expr(&ts[2..]) {
            let stmt = Stmt::Assign(box name.clone(), box expr);
            return Some((stmt, ts));
        }
    }`

Reply:
Not sure what you're referring to, but I was referring to `Box::new(...)` vs `box ...`.  I used `Box::new(...)` before I enabled `box ...`.  It's not a big deal/blocker.

Reply:
I see what you're saying now.